tests/scripts/generate_psa_tests.py

```diff
@@ -113,22 +105,44 @@ def read_implemented_dependencies(filename: str) -> FrozenSet[str]:
                      for line in open(filename)
                      for symbol in re.findall(r'\bPSA_WANT_\w+\b', line))
 _implemented_dependencies = None #type: Optional[FrozenSet[str]] #pylint: disable=invalid-name
-# This is a temporary fix for the KEY_PAIR_LEGACY symbols since they are not
-# defined in "crypto_config.h". This fix can be removed as soon as these _LEGACY
-# symbols will be removed from the code.
-_LEGACY_KEY_PAIR = ['MBEDTLS_PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_LEGACY',
-                    'MBEDTLS_PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_LEGACY']
 def hack_dependencies_not_implemented(dependencies: List[str]) -> None:
     global _implemented_dependencies #pylint: disable=global-statement,invalid-name
     if _implemented_dependencies is None:
         _implemented_dependencies = \
             read_implemented_dependencies('include/psa/crypto_config.h')
     if not all((dep.lstrip('!') in _implemented_dependencies or
-                'PSA_WANT' not in dep or
-                dep.lstrip('!') in _LEGACY_KEY_PAIR)
+                'PSA_WANT' not in dep)
                for dep in dependencies):
         dependencies.append('DEPENDENCY_NOT_IMPLEMENTED_YET')
 
+# This helper function add the proper suffix to PSA_WANT_KEY_TYPE_xxx_KEY_PAIR
+# symbols according to the required usage.
+def tweak_key_pair_dependency(dep: str, usage: str):
+    ret_list = list()
```

Comment:
Again, this is not idiomatic Python: idiomatic Python would be to make this function a generator which calls `yield` on each element.

```
def tweak_key_pair_dependency(dep: str, usage: str) -> Iterator[str]:
    if dep.endswith('RSA_KEY_PAIR'):
        yield re.sub(r'RSA_KEY_PAIR\Z', r'RSA_KEY_PAIR_LEGACY', dep)
    …
```


Let's not drag this on further. I expect I'll refactor the dependency calculation code soon and I can take care of this.